apps/mobile/src/app/_layout.tsx

```diff
@@ -51,12 +53,13 @@ export default function RootLayout() {
                   <SplashScreenGuard>
                     <GestureHandlerRootView style={{ flex: 1 }}>
                       <SheetProvider>
-                        <SendSheetProvider value={{ sendSheetRef }}>
+                        <SheetNavigatorProvider value={{ sendSheetRef, receiveSheetRef }}>
```

Comment:
Is the idea for having different named refs to allow for popping up different sheets over one another? Just wondering if it makes sense to have a single ref `fullPageSheetRef`?

Reply:
Probably does make sense to create refs per sheet for maximum flexibility.

Reply:
wdyt about moving the logic of creating the provider and the refs into a separate component so that `_layout` doesn't know anything about how that all works, it just wraps the page in another "layer" - something like `<FullsheetWrapper>`? where that looks like 

```
export default function FullsheetWrapper({ children }) {
  const sendSheetRef = useRef<SheetRef>(null);
  const receiveSheetRef = useRef<SheetRef>(null);
  
  return (<SheetNavigatorProvider value={{ sendSheetRef, receiveSheetRef }}> { children }</<SheetNavigatorProvider>);
  }```

Reply:
Yep, I can move into a wrapper, np. Prob a good idea so the _layout doesn't become bloated overtime.

Reply:
In case it helps I put `LeatherProvider` into `src/common/` recently and plan to move `ThemeProvider` + `AppRouter` into there too. 

Originally I create files directly under `app/` but Edgar said that would expose them to the router so not a good idea